File: linked_list/merge_two_sorted.py
from unittest import result
from list import LinkedList


class Solution:
    def mergeTwoLists(self, list1, list2):
        # x = list1.head
        # y=list2.head

        # while x or y:
        #     print(x.val)
        #     print(y.val)

        #     x = x.next
        #     y= y.next
        p = list1.head 
        q = list2.head
        s = None

        if not p:
            return q
        if not q:
            return p

        if p and q:
            if p.val <= q.val:
                s = p 
                p = s.next
            else:
                s = q
                q = s.next
            new_head = s 
        while p and q:
            if p.val <= q.val:
                s.next = p 
                s = p 
                p = s.next
            else:
                s.next = q
                s = q
                q = s.next
        if not p:
            s.next = q 
        if not q:
            s.next = p

        list1.head = new_head     
        return list1



llist1 = LinkedList()
llist1.bulkLoad([1])

llist2 = LinkedList()
llist2.bulkLoad([2])

ss = Solution()
result = ss.mergeTwoLists(llist1,llist2)
print(result.print_list())

File: linked_list/merge_two_sorted.java
package linked_list;

/**
 * mergei_two_sorted
 */
public class merge_two_sorted {

    public static void merge(ListNode list1, ListNode list2){

        ListNode mergedList = new ListNode();
        ListNode curr = mergedList;

        while (list1 != null && list2!= null) {

            if(list1.val < list2.val){
                curr.next = list1;
                list1 = list1.next;
            }else{
                curr.next = list2;
                list2 = list2.next;
            }
            curr = curr.next;
            
        }

        if(list1 != null){
            curr.next = list1;
        }else{
            curr.next = list2;
        }

        mergedList.next.printList();

    }

    public static void main(String[] args) {

        int[] arr1 = {1,2,4};
        ListNode head1 = ListNode.createList(arr1);
        int[] arr2 = {1,3,4};
        ListNode head2 = ListNode.createList(arr2);

        merge(head1, head2);
        
        
        
    }
    
}